Incorrect use of JUnit @TempDir annotation

**Describe the bug**
[Since JUnit 5.9.0](https://github.com/junit-team/junit5/commit/0dad5f937acd418f9a3601531b6297e88be14a61) the [`@TempDir` annotation](https://junit.org/junit5/docs/current/user-guide/#writing-tests-built-in-extensions-TempDirectory) cannot be used with `final` fields.
We are using this is a few places, eg: https://github.com/gchq/Gaffer/blob/9421d9dfed70cfa14c475b8f8d9144589bc135ab/store-implementation/proxy-store/src/test/java/uk/gov/gchq/gaffer/proxystore/integration/ProxyStoreResponseDeserialiserIT.java#L56-L57
This code causes tests to fail (when they are being run as expected - see additional context).

**To Reproduce**
Run `mvn integration-test -Dit.test=ProxyStoreResponseDeserialiserIT -pl :proxy-store` to run the IT above, it will fail with an error stating the field must not be declared as final.

**Expected behavior**
Test to not fail.

**Stack trace and errors**
```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.299 s <<< FAILURE! - in uk.gov.gchq.gaffer.proxystore.integration.ProxyStoreResponseDeserialiserIT
[ERROR] uk.gov.gchq.gaffer.proxystore.integration.ProxyStoreResponseDeserialiserIT.shouldUseOperationsResponseDeserialiserToDeserialiseOperationsResponse  Time elapsed: 0.007 s  <<< ERROR!
org.junit.jupiter.api.extension.ExtensionConfigurationException: @TempDir field [public final java.io.File uk.gov.gchq.gaffer.proxystore.integration.ProxyStoreResponseDeserialiserIT.testFolder] must not be declared as final.
```

**Additional context**
This problem has been present since #2732 was merged. However, for unknown reasons, version `3.0.0-M4` of the failsafe plugin does not run the IT above and therefore passes (it does recognise that class but none of the tests inside are run - `Tests run: 0`). Newer and older versions of this plugin **do** run the tests inside. Therefore this problem was not detected previously. In #2841 no changes are made to the tests/ITs in `proxy-store`, but this IT started being run again for unknown reasons, thereby revealing this issue related to the JUnit change. The problem of failsafe not running tests will be raised and tracked separately.

--- a/library/spark/spark-accumulo-library/src/test/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/handler/javardd/ImportJavaRDDOfElementsHandlerTest.java
+++ b/library/spark/spark-accumulo-library/src/test/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/handler/javardd/ImportJavaRDDOfElementsHandlerTest.java
@@ -56,11 +56,8 @@ public class ImportJavaRDDOfElementsHandlerTest {
 
     private static final AccumuloProperties PROPERTIES = AccumuloProperties.loadStoreProperties(StreamUtil.storeProps(ImportJavaRDDOfElementsHandlerTest.class));
 
-    @TempDir
-    static Path tempDir;
-
     @Test
-    public void checkImportJavaRDDOfElements() throws OperationException, IOException, InterruptedException {
+    public void checkImportJavaRDDOfElements(@TempDir Path tempDir) throws OperationException, IOException, InterruptedException {
         final Graph graph1 = new Graph.Builder()
                 .config(new GraphConfig.Builder()
                         .graphId("graphId")

--- a/library/spark/spark-accumulo-library/src/test/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/handler/javardd/ImportJavaRDDOfElementsHandlerTest.java
+++ b/library/spark/spark-accumulo-library/src/test/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/handler/javardd/ImportJavaRDDOfElementsHandlerTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2016-2022 Crown Copyright
+ * Copyright 2016-2023 Crown Copyright
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/library/spark/spark-accumulo-library/src/test/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/handler/javardd/ImportKeyValueJavaPairRDDToAccumuloHandlerTest.java
+++ b/library/spark/spark-accumulo-library/src/test/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/handler/javardd/ImportKeyValueJavaPairRDDToAccumuloHandlerTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2016-2022 Crown Copyright
+ * Copyright 2016-2023 Crown Copyright
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/library/spark/spark-accumulo-library/src/test/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/handler/javardd/ImportKeyValueJavaPairRDDToAccumuloHandlerTest.java
+++ b/library/spark/spark-accumulo-library/src/test/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/handler/javardd/ImportKeyValueJavaPairRDDToAccumuloHandlerTest.java
@@ -61,11 +61,8 @@ public class ImportKeyValueJavaPairRDDToAccumuloHandlerTest {
 
     private static final AccumuloProperties PROPERTIES = AccumuloProperties.loadStoreProperties(StreamUtil.storeProps(ImportKeyValueJavaPairRDDToAccumuloHandlerTest.class));
 
-    @TempDir
-    static Path tempDir;
-
     @Test
-    public void checkImportKeyValueJavaPairRDD() throws OperationException, IOException, InterruptedException {
+    public void checkImportKeyValueJavaPairRDD(@TempDir Path tempDir) throws OperationException, IOException, InterruptedException {
         final Graph graph1 = new Graph.Builder()
                 .config(new GraphConfig.Builder()
                         .graphId("graphId")

--- a/library/spark/spark-accumulo-library/src/test/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/handler/scalardd/ImportRDDOfElementsHandlerTest.java
+++ b/library/spark/spark-accumulo-library/src/test/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/handler/scalardd/ImportRDDOfElementsHandlerTest.java
@@ -57,11 +57,8 @@ public class ImportRDDOfElementsHandlerTest {
 
     private static final AccumuloProperties PROPERTIES = AccumuloProperties.loadStoreProperties(StreamUtil.storeProps(ImportRDDOfElementsHandlerTest.class));
 
-    @TempDir
-    static Path tempDir;
-
     @Test
-    public void checkImportRDDOfElements() throws OperationException, IOException {
+    public void checkImportRDDOfElements(@TempDir Path tempDir) throws OperationException, IOException {
         final Graph graph1 = new Graph.Builder()
                 .config(new GraphConfig.Builder()
                         .graphId("graphId")

--- a/library/spark/spark-accumulo-library/src/test/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/handler/scalardd/ImportRDDOfElementsHandlerTest.java
+++ b/library/spark/spark-accumulo-library/src/test/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/handler/scalardd/ImportRDDOfElementsHandlerTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2016-2022 Crown Copyright
+ * Copyright 2016-2023 Crown Copyright
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/store-implementation/proxy-store/src/test/java/uk/gov/gchq/gaffer/proxystore/integration/ProxyStoreBasicIT.java
+++ b/store-implementation/proxy-store/src/test/java/uk/gov/gchq/gaffer/proxystore/integration/ProxyStoreBasicIT.java
@@ -48,6 +48,7 @@ import uk.gov.gchq.gaffer.operation.impl.get.GetElements;
 import uk.gov.gchq.gaffer.operation.impl.job.GetJobDetails;
 import uk.gov.gchq.gaffer.operation.impl.output.ToList;
 import uk.gov.gchq.gaffer.proxystore.ProxyStore;
+import uk.gov.gchq.gaffer.proxystore.SingleUseMapProxyStore;
 import uk.gov.gchq.gaffer.rest.RestApiTestClient;
 import uk.gov.gchq.gaffer.rest.service.v2.RestApiV2TestClient;
 import uk.gov.gchq.gaffer.store.StoreTrait;
@@ -105,6 +106,7 @@ public class ProxyStoreBasicIT {
 
     @BeforeAll
     public static void beforeAll() throws Exception {
+        SingleUseMapProxyStore.cleanUp();
         CLIENT.startServer();
     }
 

--- a/store-implementation/proxy-store/src/test/java/uk/gov/gchq/gaffer/proxystore/integration/ProxyStoreBasicIT.java
+++ b/store-implementation/proxy-store/src/test/java/uk/gov/gchq/gaffer/proxystore/integration/ProxyStoreBasicIT.java
@@ -23,7 +23,6 @@ import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.io.TempDir;
 
-import uk.gov.gchq.gaffer.commonutil.CommonTestConstants;
 import uk.gov.gchq.gaffer.commonutil.StreamUtil;
 import uk.gov.gchq.gaffer.commonutil.TestGroups;
 import uk.gov.gchq.gaffer.commonutil.TestPropertyNames;
@@ -99,9 +98,6 @@ public class ProxyStoreBasicIT {
                     .build()
     };
 
-    @TempDir
-    public final File testFolder = CommonTestConstants.TMP_DIRECTORY;
-
     private Graph graph;
 
     @BeforeAll
@@ -116,7 +112,7 @@ public class ProxyStoreBasicIT {
     }
 
     @BeforeEach
-    public void before() throws IOException {
+    public void before(@TempDir File testFolder) throws IOException {
         CLIENT.reinitialiseGraph(testFolder, StreamUtil.SCHEMA, "map-store.properties");
 
         // setup ProxyStore

--- a/store-implementation/proxy-store/src/test/java/uk/gov/gchq/gaffer/proxystore/integration/ProxyStoreBasicIT.java
+++ b/store-implementation/proxy-store/src/test/java/uk/gov/gchq/gaffer/proxystore/integration/ProxyStoreBasicIT.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2017-2021 Crown Copyright
+ * Copyright 2017-2023 Crown Copyright
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/store-implementation/proxy-store/src/test/java/uk/gov/gchq/gaffer/proxystore/integration/ProxyStoreITs.java
+++ b/store-implementation/proxy-store/src/test/java/uk/gov/gchq/gaffer/proxystore/integration/ProxyStoreITs.java
@@ -16,14 +16,12 @@
 
 package uk.gov.gchq.gaffer.proxystore.integration;
 
-import org.junit.AfterClass;
 import org.junit.platform.suite.api.ConfigurationParameter;
 import org.junit.platform.suite.api.ExcludeClassNamePatterns;
 
 import uk.gov.gchq.gaffer.commonutil.StreamUtil;
 import uk.gov.gchq.gaffer.integration.AbstractStoreITs;
 import uk.gov.gchq.gaffer.proxystore.ProxyProperties;
-import uk.gov.gchq.gaffer.proxystore.SingleUseMapProxyStore;
 import uk.gov.gchq.gaffer.store.schema.Schema;
 
 import static uk.gov.gchq.gaffer.integration.junit.extensions.IntegrationTestSuiteExtension.INIT_CLASS;
@@ -32,11 +30,6 @@ import static uk.gov.gchq.gaffer.integration.junit.extensions.IntegrationTestSui
                            "uk.gov.gchq.gaffer.integration.impl.GeneratorsIT"}) // Skipped because: The output type reference doesn't deserialise the output correctly
 @ConfigurationParameter(key = INIT_CLASS, value = "uk.gov.gchq.gaffer.proxystore.integration.ProxyStoreITs")
 public class ProxyStoreITs extends AbstractStoreITs {
-
-    @AfterClass
-    public static void afterClass() {
-        SingleUseMapProxyStore.cleanUp();
-    }
 
     private static final ProxyProperties STORE_PROPERTIES = ProxyProperties
             .loadStoreProperties(StreamUtil.openStream(ProxyStoreITs.class, "/mock-proxy-store.properties"));

--- a/store-implementation/proxy-store/src/test/java/uk/gov/gchq/gaffer/proxystore/integration/ProxyStoreITs.java
+++ b/store-implementation/proxy-store/src/test/java/uk/gov/gchq/gaffer/proxystore/integration/ProxyStoreITs.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2016-2022 Crown Copyright
+ * Copyright 2016-2023 Crown Copyright
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/store-implementation/proxy-store/src/test/java/uk/gov/gchq/gaffer/proxystore/integration/ProxyStoreResponseDeserialiserIT.java
+++ b/store-implementation/proxy-store/src/test/java/uk/gov/gchq/gaffer/proxystore/integration/ProxyStoreResponseDeserialiserIT.java
@@ -30,6 +30,7 @@ import uk.gov.gchq.gaffer.operation.OperationChainDAO;
 import uk.gov.gchq.gaffer.operation.impl.add.AddElements;
 import uk.gov.gchq.gaffer.proxystore.ProxyProperties;
 import uk.gov.gchq.gaffer.proxystore.ProxyStore;
+import uk.gov.gchq.gaffer.proxystore.SingleUseMapProxyStore;
 import uk.gov.gchq.gaffer.proxystore.response.deserialiser.ResponseDeserialiser;
 import uk.gov.gchq.gaffer.rest.RestApiTestClient;
 import uk.gov.gchq.gaffer.rest.service.v2.RestApiV2TestClient;
@@ -58,6 +59,7 @@ public class ProxyStoreResponseDeserialiserIT {
 
     @BeforeAll
     public static void beforeAll() {
+        SingleUseMapProxyStore.cleanUp();
         CLIENT.startServer();
     }
 

--- a/store-implementation/proxy-store/src/test/java/uk/gov/gchq/gaffer/proxystore/integration/ProxyStoreResponseDeserialiserIT.java
+++ b/store-implementation/proxy-store/src/test/java/uk/gov/gchq/gaffer/proxystore/integration/ProxyStoreResponseDeserialiserIT.java
@@ -21,7 +21,6 @@ import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.io.TempDir;
 
-import uk.gov.gchq.gaffer.commonutil.CommonTestConstants;
 import uk.gov.gchq.gaffer.commonutil.StreamUtil;
 import uk.gov.gchq.gaffer.graph.Graph;
 import uk.gov.gchq.gaffer.operation.Operation;
@@ -54,9 +53,6 @@ public class ProxyStoreResponseDeserialiserIT {
 
     private static final RestApiTestClient CLIENT = new RestApiV2TestClient();
 
-    @TempDir
-    public final File testFolder = CommonTestConstants.TMP_DIRECTORY;
-
     @BeforeAll
     public static void beforeAll() {
         SingleUseMapProxyStore.cleanUp();
@@ -69,7 +65,7 @@ public class ProxyStoreResponseDeserialiserIT {
     }
 
     @BeforeEach
-    public void before() throws IOException {
+    public void before(@TempDir File testFolder) throws IOException {
         CLIENT.reinitialiseGraph(testFolder, StreamUtil.SCHEMA, "map-store.properties");
     }
 

--- a/store-implementation/proxy-store/src/test/java/uk/gov/gchq/gaffer/proxystore/integration/ProxyStoreResponseDeserialiserIT.java
+++ b/store-implementation/proxy-store/src/test/java/uk/gov/gchq/gaffer/proxystore/integration/ProxyStoreResponseDeserialiserIT.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2021 Crown Copyright
+ * Copyright 2021-2023 Crown Copyright
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -13,6 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 package uk.gov.gchq.gaffer.proxystore.integration;
 
 import org.junit.jupiter.api.AfterAll;

